File: android/app/src/main/java/fr/gshz/hideandseek/domain/model/ZonePlacement.kt
package fr.gshz.hideandseek.domain.model

/**
 * Where the hiding team is settling: the station point, the radius, and the name of the station the
 * hider tapped, read off the transit overlay so Move can name it to the seekers later.
 */
data class ZonePlacement(
    val lat: Double,
    val lng: Double,
    val radiusMeters: Double? = null,
    val stationName: String? = null,
)

File: backend/src/Dto/ZonePlacement.php
<?php

declare(strict_types=1);

namespace App\Dto;

use LongitudeOne\Spatial\PHP\Types\Geography\Point;

/**
 * Where a hiding team is settling: the station point, the radius, and the station name.
 * The name comes from the client: its transit overlay is the same LOOM output the server
 * imported, so the hider names the station they tapped instead of the server guessing.
 */
final readonly class ZonePlacement
{
    public function __construct(
        public Point $point,
        public ?float $radiusMeters = null,
        public ?string $stationName = null,
    ) {
    }
}
